web/src/filter.ts

```diff
@@ -1128,6 +1128,24 @@ export class Filter {
         if (this.single_term_type_returns_all_messages_of_conversation()) {
             return true;
         }
+
+        const term_types = this.sorted_term_types();
+
+        if (_.isEqual(term_types, ["not-is-dm"])) {
+            return true;
+        }
```

Comment:
It seems like at least this type of rule should be in `single_term_type_returns_all_messages_of_conversation`?